typed-racket-lib/typed-racket/types/update.rkt

```diff
@@ -57,6 +57,14 @@
             [(Bottom:) -Bottom]
             [ty (let ([flds (append lhs (cons (make-fld ty acc-id #f) rhs))])
                   (make-Struct nm par flds proc poly pred))])]
+
+         ;; prefab struct ops
+         [((Prefab: key flds) (PrefabPE: path-key idx))
```

Comment:
We're assuming we can't have paths to mutable fields so we can't get here in that case.